packages/ui/components/form-core/src/choice-group/CustomChoiceGroupMixin.js

```diff
@@ -0,0 +1,198 @@
+        // @ts-ignore
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // @ts-ignore
+        this._customChoices = new Set(ensureArray(value));
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    // @ts-ignore
+    get formattedValue() {
+      return this.__getChoicesFrom(super.formattedValue);
+    }
+
+    set formattedValue(value) {
+      super.formattedValue = value;
+
+      if (value === null || value === undefined) {
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // Convert formattedValue to modelValue to store as custom choices, or fall back to the input value
+        this._customChoices = new Set(
+          ensureArray(value).map(
+            val => this.formElements.find(el => el.formattedValue === val)?.modelValue || val,
+          ),
+        );
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    // @ts-ignore
+    get serializedValue() {
+      return this.__getChoicesFrom(super.serializedValue);
+    }
+
+    set serializedValue(value) {
+      super.serializedValue = value;
+
+      if (value === null || value === undefined) {
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // Convert serializedValue to modelValue to store as custom choices, or fall back to the input value
+        this._customChoices = new Set(
+          ensureArray(value).map(
+            val => this.formElements.find(el => el.serializedValue === val)?.modelValue || val,
+          ),
+        );
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    /**
+     * Custom elements are all missing elements that have no corresponding element, independent if enabled or not.
+     */
+    // @ts-ignore
+    get customChoices() {
+      if (!this.allowCustomChoice) {
+        return [];
+      }
+
+      const elems = this._getCheckedElements();
+
+      return Array.from(this._customChoices).filter(
+        choice => !elems.some(elem => elem.choiceValue === choice),
+      );
+    }
+
+    constructor() {
+      super();
+
+      this.allowCustomChoice = false;
+
+      /**
+       * @type {Set<unknown>}
+       * @protected
+       */
+      this._customChoices = new Set();
+    }
+
+    /**
+     * @private
+     */
+    // @ts-ignore
+    __getChoicesFrom(input) {
+      const values = input;
+      if (!this.allowCustomChoice) {
+        return values;
+      }
+
+      if (this.multipleChoice) {
+        return [...ensureArray(values), ...this.customChoices];
+      }
+
+      if (values === '') {
+        return this._customChoices.values().next().value || '';
+      }
+
+      return values;
+    }
+
+    /** @param {import('lit').PropertyValues} changedProperties */
+    updated(changedProperties) {
```

Comment:
Could you elaborate on what happens here exactly?

(If I delete the `updated` callback, the tests still run)

Reply:
I've added to fix a issue I had where it was not triggering the events properly, but indeed it's not necessary now. Might have solved while cleaning up the code at some point and forgot to remove it.